index.bs

```diff
@@ -1149,9 +1149,35 @@ The [=new=] {{MLGraphBuilder}} constructor steps are:
 1. If [=this=]'s [=relevant global object=]'s [=associated Document=] is not [=allowed to use=] the [=webnn-feature|webnn=] feature, throw a "{{SecurityError}}" {{DOMException}} and abort these steps.
 1. Let |context| be the first argument.
 1. If the <a>validate MLContext</a> steps given |context| return `false`, throw a "{{TypeError}}" and abort these steps.
-
 1. Set {{MLGraphBuilder/[[context]]}} to |context|.
 
+### The {{MLGraphBuilder/input()}} method  ### {#api-mlgraphbuilder-input}
+Create a named {{MLOperand}} based on a descriptor, that can be used as an input.
+<div class="note">
+    **Arguments:**
+        - *name*: a [=string=] name of the input.
+        - *descriptor*: an {{MLOperandDescriptor}} object.
+    **Returns:**: an {{MLOperand}} object.
+</div>
+<div algorithm=input>
+    The {{MLGraphBuilder/input(name, descriptor)}} steps are:
+    <div class="note">
+        The permissions and context validity have been checked by [[#api-mlgraphbuilder-constructor]] steps.
+    </div>
+    1. Let |name| be the first argument.
+        1. If |name| is `undefined` or an empty [=string=], then throw a "{{TypeError}}" {{DOMException}} and stop.
+    1. Let |descriptor| be the second argument.
+        1. If |descriptor| is not an an object that [=implements=] {{MLOperandDescriptor}}, then throw a "{{TypeError}}" {{DOMException}} and stop.
+        1.  If the [=byte length=] of |descriptor| is not supported by the underlying platform, then throw a "{{DataError}}" {{DOMException}} and stop.
+        1. If the [=check dimensions=] steps given |descriptor|.{{MLOperandDescriptor/type}} and |descriptor|.{{MLOperandDescriptor/dimensions}} return `false`, throw a "{{DataError}}" {{DOMException}} and stop.
```

Comment:
What would happen if user code passes a descriptor for scalar operand (`dimensions` is undefined)? This `input()` algorithm seems only to support tensor operand now.

Reply:
Actually, the `constant(descriptor, bufferView)` algorithm in PR #365 should also address this issue.

Reply:
Right. So this is where conventions about `descriptor` become important.
I am thinking if it's enough to distinguish between vector and scalar solely on the existence of a `dimensions` member. But for now let's continue with that working assumption and keep it in mind.

About `constant(descriptor, bufferView)`: the `descriptor.dimensions` must be defined, otherwise it's an error. There is a polymorphic variant for specifying scalar constant. Should we change that?

Reply:
Done, please check if correct. Using an assert seemed better choice than just making a note.
Forgot to fix the white space error, such a small change that I amended the same commit and force-pushed.

Reply:
> About constant(descriptor, bufferView): the descriptor.dimensions must be defined, otherwise it's an error.

That's fine. No change needed.